src/sockperf.cpp

```diff
@@ -2304,6 +2323,21 @@ void cleanup() {
                 if (g_fds_array[ifd]->is_multicast) {
                     FREE(g_fds_array[ifd]->memberships_addr);
                 }
+                if (s_user_params.addr.ss_family == AF_UNIX) {
+                    unlink(s_user_params.client_bind_info.addr_un.sun_path);
+                    if (s_user_params.mode == MODE_CLIENT && s_user_params.sock_type == SOCK_DGRAM) { // unlink binded client
+                        std::string sun_path = s_user_params.addr.addr_un.sun_path;
+                        char pid_buffer [MAX_FDS_NUM];
+                        char ifd_buffer [MAX_FDS_NUM];
+                        snprintf(pid_buffer, sizeof(pid_buffer), "%d", getpid());
+                        snprintf(ifd_buffer, sizeof(ifd_buffer), "%d", ifd);
+                        sun_path = sun_path + "_" + pid_buffer + "_" + ifd_buffer;
+                        log_dbg("unlinking %s",sun_path.c_str());
+                        unlink(sun_path.c_str());
+                    }
+                    if (s_user_params.mode == MODE_SERVER)
+                        unlink(g_fds_array[ifd]->server_addr.addr_un.sun_path);
```

Comment:
Will cause out-of-buffer-read if socket name is `/tmp/socket00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000` (MAX_UDS_NAME symbols long)

Reply:
but wouldn't we fail much earilier in `resolve_sockaddr` stage, when there is a check if 
`path.length() > MAX_UDS_NAME`?

Reply:
I'm talking about `path.length() == MAX_UDS_NAME`. Probably the best solution is to check length **including** trailing zero and force the last symbol in buffer to be zero.

Reply:
Why not just the condition from `path.length() > MAX_UDS_NAME` to `path.length() >= MAX_UDS_NAME`?

Reply:
Yes, that's exactly what I mean.

Reply:
can we close this one?